pallets/automation-time/src/lib.rs

```diff
@@ -796,22 +801,16 @@ pub mod pallet {
 			<T as Config>::WeightInfo::run_native_transfer_task()
 		}
 
-		/// Determines if a task has been completed.
-		/// 
-		/// A task has been completed if all execution times exist in the past and/or are in the current time slot.
-		/// This means there a no more execution times to be run.
-		fn is_completed_task(mut execution_times: Vec<UnixTime>) -> bool {
-			if execution_times.len() == 0 { return true };
-
-			let current_time_slot = match Self::get_current_time_slot() {
-				Ok(time_slot) => time_slot,
-				// This will only occur for the first block in the chain.
-				Err(_) => return false,
-			};
-
-			Self::clean_execution_times_vector(&mut execution_times);
-			let is_complete = *execution_times.last().unwrap() <= current_time_slot;
-			return is_complete;
+		/// Decrements task executions left.
+		/// If task is complete then removes task. If task not complete update task map.
+		/// A task has been completed if executions left equals 0.
+		fn decrement_task_and_remove_if_complete(task_id: T::Hash, mut task: Task<T>) {
+			task.executions_left = task.executions_left - 1;
+			if task.executions_left == 0 {
+				Tasks::<T>::remove(task_id);
+			} else {
+				Tasks::<T>::insert(task_id, task);
```

Comment:
prob test to make sure that this actually overwrites that task on the map. almost certain should, but check to make sure